cmd/checksum.go

```diff
@@ -0,0 +1,106 @@
+// Copyright © 2017 Prometheus Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	checksumsFilename = "sha256sums.txt"
+)
+
+// checksumCmd represents the command calculating checksums for all files in a
+// given directory and writing the result to a checksums file.
+var checksumCmd = &cobra.Command{
+	Use:   "checksum [<location>]",
+	Short: "Calculate the SHA256 checksum for each file in the given location",
+	Long:  `Calculate the SHA256 checksum for each file in the given location`,
+	Run: func(cmd *cobra.Command, args []string) {
+		runChecksum(optArg(args, 0, "."))
+	},
+}
+
+// init prepares cobra flags
+func init() {
+	Promu.AddCommand(checksumCmd)
+}
+
+func runChecksum(path string) {
+	checksums, err := calculateSHA256s(path)
+	if err != nil {
+		fatalMsg("Failed to calculate checksums", err)
+	}
+
+	file, err := os.Create(filepath.Join(path, checksumsFilename))
+	if err != nil {
+		fatalMsg("Failed to create checksums file", err)
+	}
+	for _, c := range checksums {
+		if _, err := fmt.Fprintf(file, "%x  %s\n", c.checksum, c.filename); err != nil {
+			fatalMsg("Failed to write to checksums file", err)
+		}
+	}
+	file.Close()
```

Comment:
Leftover from an earlier version (where I just printed the output to stdout). It doesn't really matter here, as we'd exit and release the fd anyway immediately, but changed it as you're generally right.